### PR TITLE
[CELEBORN-185][SPARK] Can't release shuffle data if rss fallback to nss

### PR DESCRIPTION
**Pull request by jxysoft**
### What changes were proposed in this pull request?
add shuffle id to sortShuffleIds while getWriter if rss fallback to native shuffle service

### Why are the changes needed?
if rss fallback to native shuffle service, RssShuffleManager.unregisterShuffle won't call SortShuffleManager to release shuffle data on executor size

### CODE REVIEW

**Comment by pan3793** [comment]
What's "nss"? Do you mean ESS (External Shuffle Service)?

**Comment by waitinfuture** [comment]
> What's "nss"? Do you mean ESS (External Shuffle Service)?

AFAIK nss means no shuffle service, standalone.

**Comment by pan3793** [comment]
> > What's "nss"? Do you mean ESS (External Shuffle Service)?
> 
> AFAIK nss means no shuffle service, standalone.

Got it, thanks for the explaination.

File: client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/RssShuffleManager.java [hunk 1]
```diff
@@ -172,6 +172,7 @@ public <K, V> ShuffleWriter<K, V> getWriter(
               "Unrecognized shuffle write mode!" + celebornConf.shuffleWriterMode());
         }
       } else {
+        sortShuffleIds.add(shuffleId);
```
Comment (waitinfuture) [inline client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/RssShuffleManager.java:175]: should be
```
sortShuffleIds.add(handle.shuffleId());
```

File: client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/RssShuffleManager.java [hunk 2]
```diff
@@ -170,6 +170,7 @@ public <K, V> ShuffleWriter<K, V> getWriter(
               "Unrecognized shuffle write mode!" + celebornConf.shuffleWriterMode());
         }
       } else {
+        sortShuffleIds.add(shuffleId);
```
Comment (waitinfuture) [inline client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/RssShuffleManager.java:173]: ditto

**Comment by jxysoft** [comment]
> What's "nss"? Do you mean ESS (External Shuffle Service)?

native shuffle service

**Comment by waitinfuture** [comment]
> > What's "nss"? Do you mean ESS (External Shuffle Service)?
> 
> native shuffle service



> > What's "nss"? Do you mean ESS (External Shuffle Service)?
> 
> native shuffle service

what does native shuffle service mean?

File: client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/RssShuffleManager.java [hunk 3]
```diff
@@ -172,6 +172,7 @@ public <K, V> ShuffleWriter<K, V> getWriter(
               "Unrecognized shuffle write mode!" + celebornConf.shuffleWriterMode());
         }
       } else {
+        sortShuffleIds.add(handle.shuffleId);
```
Comment (waitinfuture) [inline client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/RssShuffleManager.java:175]: seems we have to add parenthesis after shuffleId
```
handle.shuffleId()
```

File: client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/RssShuffleManager.java [hunk 4]
```diff
@@ -170,6 +170,7 @@ public <K, V> ShuffleWriter<K, V> getWriter(
               "Unrecognized shuffle write mode!" + celebornConf.shuffleWriterMode());
         }
       } else {
+        sortShuffleIds.add(handle.shuffleId);
```
Comment (waitinfuture) [inline client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/RssShuffleManager.java:173]: ditto

File: client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/RssShuffleManager.java (hunk 1 above)
Comment (jxysoft) [reply client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/RssShuffleManager.java]: thanks, fixed

File: client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/RssShuffleManager.java (hunk 3 above)
Comment (jxysoft) [reply client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/RssShuffleManager.java]: thanks, fixed

**Comment by jxysoft** [comment]
>

**Comment by jxysoft** [comment]
> > > What's "nss"? Do you mean ESS (External Shuffle Service)?
> > 
> > 
> > native shuffle service
> 
> > > What's "nss"? Do you mean ESS (External Shuffle Service)?
> > 
> > 
> > native shuffle service
> 
> what does native shuffle service mean?

spark built-in shuffle implementation,   not ESS

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Comment by AngersZhuuuu** [comment]
@waitinfuture Should we backport this to branch-0.2? and release in branch-0.2.1?

**Comment by pan3793** [comment]
> @waitinfuture Should we backport this to branch-0.2? and release in branch-0.2.1?

It's already there and will be included in 0.2.0-incubating.

**Pull request merged**